src/test/java/de/unijena/cheminf/clustering/desccalc/DescriptorTest.java

```diff
@@ -134,49 +167,73 @@ public void test_WIENER_NUMBER() throws Exception {
             Assertions.assertEquals(2, Descriptor.getNumberOfComponents(tmpDescriptors));
 
             float[][] tmpMatrix = new float[][]
-                {
-                    {0f, 0f}
-                };
+                    {
+                            {0f, 0f}
+                    };
+            List<int[]> aNanPositions = Collections.synchronizedList(new LinkedList<>());
+            Assertions.assertTrue(
+                    Descriptor.setDescriptorsForMoleculesByMoleculeParallelizationSynchronized(
+                            tmpDescriptors,
+                            tmpMoleculesArray,
+                            tmpMatrix,
+                            tmpStartIndex,
+                            tmpIsParallelCalculation,
+                            aNanPositions
+                    )
+            );
+            Assertions.assertEquals("9", tmpFormat.format(tmpMatrix[0][0])); // 1(C1C2)+2(C1O1)+2(C1O2)+1(C2O1)+1(C2O2)+2(O1O2) = 9
+            Assertions.assertEquals("0", tmpFormat.format(tmpMatrix[0][1])); // there are no atoms that are 3 bonds apart
```

Comment:
These are integers that should be compared as such. Same is true for other test methods.